util/scripts/backup-existing-target.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+# SPDX-FileCopyrightText: 2022 Mischback
+# SPDX-License-Identifier: MIT
+# SPDX-FileType: SOURCE
+
+# Create a backup of an existing file.
+#
+# @param: The path/name of the symlink, relative to the user's home directory
+# @param: The target of the link, relative to this script's directory (which
+#         should be the repository root)
+#
+# The function will remove existing symlinks and create backups of existing
+# files and directories before creating the actual symlink.
```

Comment:
Document why this file is not executable (e.g. it is only included from other scripts).